### PR TITLE
SciPy 1.13.0 rc's

### PR DESCRIPTION
**Pull request by h-vetinari**

### CODE REVIEW

**Comment by h-vetinari** [comment]
Ugh, the macro pollution by the windows SDK is just blergh:
```
[64/1477] Compiling C object scipy/special/lib_amos.a.p/_amos.c.obj
FAILED: scipy/special/lib_amos.a.p/_amos.c.obj 
"clang-cl" "-Iscipy\special\lib_amos.a.p" "-Iscipy\special" "-I..\scipy\special" "-Iscipy\_lib" "-I..\scipy\_lib" "-I..\scipy\_build_utils\src" "-DNDEBUG" "/MD" "/nologo" "/showIncludes" "/utf-8" "/W2" "/clang:-std=c99" "/O2" "-Wno-unused-but-set-variable" "-Wno-unused-function" "-Wno-conversion" "-Wno-misleading-indentation" "-D_USE_MATH_DEFINES" "/Fdscipy\special\lib_amos.a.p\_amos.c.pdb" /Foscipy/special/lib_amos.a.p/_amos.c.obj "/c" ../scipy/special/_amos.c
In file included from ../scipy/special/_amos.c:96:
../scipy/special\_amos.h(106,15): error: expected ';' after top level declarator
  106 | double complex amos_airy(double complex, int, int, int *, int *);
      |               ^
      |               ;
../scipy/special\_amos.h(107,61): error: redefinition of parameter '_complex'
  107 | int amos_besh(double complex, double, int, int, int, double complex *, int *);
      |                                                             ^
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\corecrt_math.h(44,29): note: expanded from macro 'complex'
   44 |             #define complex _complex
      |                             ^
```

**Comment by h-vetinari** [comment]
On OSX, we run into the exact same problem as for [numpy](https://github.com/conda-forge/numpy-feedstock/pull/312) (both vendoring pocketfft):
```
[881/1477] Compiling C++ object scipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p/pypocketfft.cxx.o
FAILED: scipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p/pypocketfft.cxx.o 
x86_64-apple-darwin13.4.0-clang++ -Iscipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p -Iscipy/fft/_pocketfft -I../scipy/fft/_pocketfft -I../scipy/_lib/pocketfft -I$PREFIX/include -I$PREFIX/include/python3.10 -fvisibility=hidden -fvisibility-inlines-hidden -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c++17 -O3 -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/scipy-split-1.13.0rc1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D_FORTIFY_SOURCE=2 -isystem $PREFIX/include -mmacosx-version-min=10.9 -DPOCKETFFT_PTHREADS -DPOCKETFFT_CACHE_SIZE=16 -MD -MQ scipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p/pypocketfft.cxx.o -MF scipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p/pypocketfft.cxx.o.d -o scipy/fft/_pocketfft/pypocketfft.cpython-310-darwin.so.p/pypocketfft.cxx.o -c ../scipy/fft/_pocketfft/pypocketfft.cxx
In file included from ../scipy/fft/_pocketfft/pypocketfft.cxx:19:
../scipy/_lib/pocketfft/pocketfft_hdronly.h:160:15: error: no member named 'aligned_alloc' in the global namespace; did you mean simply 'aligned_alloc'?
  void *ptr = ::aligned_alloc(align,(size+align-1)&(~(align-1)));
              ^~~~~~~~~~~~~~~
              aligned_alloc
../scipy/_lib/pocketfft/pocketfft_hdronly.h:157:14: note: 'aligned_alloc' declared here
inline void *aligned_alloc(size_t align, size_t size)
             ^
1 error generated.
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% set version = "1.12.0" %}
+{% set version = "1.13.0rc1" %}
```
Comment (h-vetinari) [inline recipe/meta.yaml:1]:
```suggestion
{% set version = "1.13.0" %}
```

**Comment by rgommers** [comment]
Closing in preference of gh-271.

**Pull request closed without merging**